packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,381 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import 'terra-base/lib/baseStyles';
+import Field from 'terra-form/lib/Field';
+import Fieldset from 'terra-form/lib/Fieldset';
+import NumberField from 'terra-form/lib/NumberField';
+import DatePicker from 'terra-date-picker';
+import SelectField from 'terra-form/lib/SelectField';
+import OnsetUtils from './OnsetUtils';
+
+const propTypes = {
+  /**
+   * The ISO 8601 string representation of the birth date to calculate an onset date for the AGE precision.
+   * Also limits the earliest possible date that can be selected for an onset date for YEAR, MONTH, and DATE precision.
+   */
+  birthdate: PropTypes.string.isRequired,
+
+  /**
+   * The granularity of the onset date. AGE, YEAR, MONTH, and DATE are accepted.
+   */
+  granularity: PropTypes.oneOf(['AGE', 'YEAR', 'MONTH', 'DATE']),
+
+  /**
+   * Name of the granularity select. The name should be unique.
+   */
+  granularitySelectName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a granularity is selected.
+   * The first parameter is the changed granularity value.
+   */
+  granularitySelectOnChange: PropTypes.func,
+
+  /**
+   * The precision of the onset date. This should be one of precisions passed to the precisionSet prop.
+   */
+  precision: PropTypes.oneOf([
+    OnsetUtils.PrecisionOptions.ONAT,
+    OnsetUtils.PrecisionOptions.ABOUT,
+    OnsetUtils.PrecisionOptions.BEFORE,
+    OnsetUtils.PrecisionOptions.AFTER,
+    OnsetUtils.PrecisionOptions.UNKNOWN,
+  ]),
+
+  /**
+   * The set of precisions that can be used with the onset picker.
+   * One of OnsetPicker.PrecisionOptions.ONAT, OnsetPicker.PrecisionOptions.ABOUT, OnsetPicker.PrecisionOptions.BEFORE,
+   * OnsetPicker.PrecisionOption.AFTER, OnsetPicker.PrecisionOptions.UNKNOWN.
+   * Order of precisions determines order in precision select.
+   */
+  precisionSet: PropTypes.arrayOf(PropTypes.oneOf([
+    OnsetUtils.PrecisionOptions.ONAT,
+    OnsetUtils.PrecisionOptions.ABOUT,
+    OnsetUtils.PrecisionOptions.BEFORE,
+    OnsetUtils.PrecisionOptions.AFTER,
+    OnsetUtils.PrecisionOptions.UNKNOWN,
+  ])),
+
+  /**
+   * Name of the precision select. The name should be unique.
+   */
+  precisionSelectName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a precision is selected.
+   * The first parameter is the changed precision value.
+   */
+  precisionSelectOnChange: PropTypes.func,
+
+  /**
+   * The ISO 8601 string representation of the onset date to view/modify. Defaults to current date.
+   */
+  onsetDate: PropTypes.string,
+
+  /**
+   * Name of the onsetDate input. The name should be unique.
+   */
+  onsetDateInputName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a onsetDate is changed.
+   * The first parameter is the changed onsetDate value.
+   */
+  onsetDateInputOnChange: PropTypes.func,
```

Comment:
I think the parameter passed in this prop should be in ISO 8601 format instead of a moment object so that the format is consistent with the other props and consumers are not required to pull in the momentjs lib to parse it.

Reply:
I know initially I was returning it as a string, I'm unsure when I changed that. I'm a tad upset that my tests were not broken by this change...

Reply:
Now returning formatted string instead of moment object